### DISCUSSION TITLE
ZuluSCSI RP2040 does not recognize SD card formatted in FAT32

### DISCUSSION THREAD

**Discussion by Br0ther66** [ZuluSCSI/ZuluSCSI-firmware#235] [General]
Good afternoon.
I recently purchased a zuluscsi rp2040 for one of the best SCSI controllers ever made for the Commodore Amiga 500, the Supradrive 500XP controller. At first everything worked fine with the SD card formatted in FAT32 and the copied images. I tried to write a raw image to the SD card to see how it worked and it seemed to me that it was very slow. I wanted to go back to how it was at the beginning but zuluscsi rp2040 does not recognize any SD card perfectly formatted in fat 32 and with the image files copied to it, image files that previously had worked fine. The red led flashes 5 times, indicating that it does not recognize any connected SD card. On the other hand, if I write a raw image to the SD card again, the led blinks 3 times and it boots the Amiga's 3.1 operating system, but it reads and writes terribly slow. Is there a way to get everything back to how it was originally? Or has the ZuluSCSI RP2040 suffered some damage? I'm totally lost and I don't know what to do. I hope there is a solution, thank you very much in advance.

Excuse me for my bad English.
all the best
Fran.

**Comment by PetteriAimonen** [comment]
Hi Fran,

It's unlikely there is any damage to the ZuluSCSI.

The behavior sounds like a problem with the SD card. Have you tried different cards?

The problem can also be insufficient power supply through the SCSI cable. You can try connecting with USB cable or the 4-pin connector to provide more power.

When connected to PC with USB cable, you can use a serial terminal program (for example [CoolTerm](https://learn.sparkfun.com/tutorials/terminal-basics/coolterm-windows-mac-linux) to connect to ZuluSCSI. Then you will see any error messages even if they cannot be written to the SD card.

**Reply by Br0ther66** [reply]
Thanks for the reply Petteri.
I had previously read about what you say about the power supply, but that's not what it's about. Even so, I did tests with another more powerful power supply (a PC one) and the result is the same. And as I said at the beginning, before it worked perfectly.

The current SD card that I use is 32GB and in Windows it has no problems, it formats correctly and the files are recorded without errors. Would a smaller capacity card work, for example a 2GB SD card? If the answer is yes, it is already tested and the behavior is the same.

Do you know how I can connect to PC with USB cable? What I need? where does it connect to the RP2040? I hope that the terminal program sheds some light, the led indicator is limited to saying that there is no card inserted and if the terminal answers the same I am lost.

Thanks in advance Petteri
Fran.

**Comment by aperezbios** [comment]
Francisco, 

Please see my response to your e-mail, as I think I know what's going on here. Engaging simultaneously via multiple support channels makes it more difficult for us to communicate effectively with you.

**Reply by Br0ther66** [reply]
Thanks Alex, and excuse me, from now on we continue here, if that's okay with you.

I will do what you recommend about reformatting it with the official SD Card Association formatter. Although I do not understand very well about having to format it like this, I had previously formatted it with windows in fat 32 and I have copied image files that I know have previously worked well.

Now the led flashes 5 times over and over again with the card inserted.

Please be a little patient with me, I don't speak english very well and some things escape me. I don't understand what "clobber" means in the email you sent me.

Thanks for everything.
Fran

**Reply by aperezbios** [reply]
@Br0ther66 that's fine with me :) You can substitute "overwrite" for "clobber". As I stated in my e-mail, I believe the firmware on your board entered raw passthrough mode, without you necessarily realizing it. If you wrote to it in that state from the AmigaOS side, you would have over-written the FAT filesystem with an Amiga-style RDB+FFS (or whatever other filesystem you're using) which would lead to the SD card becoming unreadable on a modern PC.

**Reply by Br0ther66** [reply]
Thanks Alex. Now I am at work, when I get home I will format the card with the utility that you have said. I'll let you know how everything went.
Thank you very much for the help!

All the best

**Comment by Br0ther66** [comment]
Yourself are fantastic! Thanks for guiding me in the right direction. A couple of days ago I tried to use 'Empty File Creator' but it doesn't work in my old Windows XP, however 'SD Memory Card Formatter 5.0.2' works wonderfully. It is clear that Windows formatting is not enough. Everything has returned to normal as when I started.

My Zulu Log
```
[10ms] Platform: ZuluSCSI RP2040
[10ms] FW Version: 23.03.30-release Mar 30 2023 16:53:18
[11ms] DIP switch settings: debug log 0, termination 1
[11ms] SCSI termination is enabled
[12ms] Flash chip size: 4096 kB
[14ms] SCSI target/disk mode selected by DIP switch, acting as a SCSI disk
[20ms] SD card detected, FAT32 volume size: 29568 MB
[20ms] SD MID: 0x27, OID: 0x50 0x48
[21ms] SD Name: SD32G
[21ms] SD Date: 11/2022
[21ms] SD Serial: 0x0165625A
[22ms] Config file zuluscsi.ini not found, using defaults
[22ms] Active configuration:
[23ms] -- SelectionDelay: 255
[23ms] -- EnableSCSI2 is on
[23ms] -- EnableParity is on
[30ms] Finding HDD images in directory /:
[30ms] -- Opening /HD0.hda for id:0 lun:0
[366ms] ---- Read prefetch enabled: 8192 bytes
[366ms] -- Opening /HD1.hda for id:1 lun:0
[915ms] ---- Read prefetch enabled: 8192 bytes
[915ms] -- Opening /HD2.hda for id:2 lun:0
[1457ms] ---- Read prefetch enabled: 8192 bytes
[1457ms] -- Platform supports ROM drive up to 3740 kB
[1458ms] ---- ROM drive image not detected
[1458ms] SCSI ID:0 BlockSize:512 Type:0 Quirks:0 ImageSize:2097152kB
[1459ms] SCSI ID:1 BlockSize:512 Type:0 Quirks:0 ImageSize:3407872kB
[1460ms] SCSI ID:2 BlockSize:512 Type:0 Quirks:0 ImageSize:3407872kB
[1560ms] Initialization complete!
```


SelectionDelay: 255 I don't understand very well what he does. The 255 value will slow the start of the Amiga OS? I notice that something is going. Maybe it is also that my SCSI controller does not give more.
Would you recommend me to update to the last firmware?

Thank you very much for all your help Petteri, Alex !!

All the best from Seville (Spain)
Fran

**This is my super Amiga 500...**
<img src="https://i.ibb.co/Y7LkRvH/Amiga-500-01.jpg" alt="Amiga-500-01" border="0">
<img src="https://i.ibb.co/4ZnDTgt/Amiga-500-02.jpg" alt="Amiga-500-02" border="0">

**All the inventions that I have been adding**
<img src="https://i.ibb.co/tz5CB6B/Amiga-500-03.jpg" alt="Amiga-500-03" border="0">

**68020 or 68000 (to choose)**
<img src="https://i.ibb.co/vvMgvGk/Amiga-500-04.jpg" alt="Amiga-500-04" border="0">
<img src="https://i.ibb.co/Vv1CBZB/Amiga-500-05.jpg" alt="Amiga-500-05" border="0">
<img src="https://i.ibb.co/tpqqL9L/Amiga-500-06.jpg" alt="Amiga-500-06" border="0">

**Switch 1 (3 Pos) : 68000 - 68020 - 68020 + 4MB Fast RAM 32 Bits
Switch 2: ROM 1.3 - ROM 3.1
Switch 3: 512k Chip Mem (On Trap Door) - Or 512k Slow Fast Men (On Trap Door)
Switch 4: 1MB Chip Mem (Megachip Card) - Or 512k Chip Mem (Main Board)
Switch 3 & 4 ON: 2MB Chip Mem**
<img src="https://i.ibb.co/m6qx1r1/Amiga-500-07.jpg" alt="Amiga-500-07" border="0">

**External Disk Drive (Device DF1) can be changed to Device DF0 (ideal for Gotek External)**
<img src="https://i.ibb.co/zrCW957/Amiga-500-08.jpg" alt="Amiga-500-08" border="0">
<img src="https://i.ibb.co/xMwSSNP/Amiga-500-09.jpg" alt="Amiga-500-09" border="0">
<img src="https://i.ibb.co/722CbhH/Amiga-500-10.jpg" alt="Amiga-500-10" border="0">

**Original Commodore Monitor 1085s Recently acquired**
<img src="https://i.ibb.co/0Zdr8ZN/Amiga-500-11.jpg" alt="Amiga-500-11" border="0">

**External CD ROM or CDTV A570 working**
<img src="https://i.ibb.co/4tF3s8L/Amiga-500-12.jpg" alt="Amiga-500-12" border="0">

**Supradrive External Fast RAM Select - 2MB (To make it work Action Replay MKIII & 68000, 3.1 ROM & Zulu SCSI) - 4MB - 8MB**
<img src="https://i.ibb.co/JcsHZpy/Amiga-500-13.jpg" alt="Amiga-500-13" border="0">
<img src="https://i.ibb.co/FHqm3R9/Amiga-500-14.jpg" alt="Amiga-500-14" border="0">
<img src="https://i.ibb.co/tQbvsPY/Amiga-500-15.jpg" alt="Amiga-500-15" border="0">
<img src="https://i.ibb.co/wNJPzp8/Amiga-500-16.jpg" alt="Amiga-500-16" border="0">

**Thus is hidden**
<img src="https://i.ibb.co/mNXzm0T/Amiga-500-17.jpg" alt="Amiga-500-17" border="0">

**And my precious recent acquisition ZuluSCSI RP2040**
<img src="https://i.ibb.co/jzXNvbb/Amiga-500-18.jpg" alt="Amiga-500-18" border="0">
